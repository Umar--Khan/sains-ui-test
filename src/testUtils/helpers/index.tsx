import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import { BasketContext, BasketWrapper } from 'src/context/BasketContext';

export const withRouter = (children: any) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export const renderWithBasketContext = (value: BasketWrapper, children: any) =>
  render(
    <BasketContext.Provider value={value}>
      {withRouter(children)}
    </BasketContext.Provider>,
  );
