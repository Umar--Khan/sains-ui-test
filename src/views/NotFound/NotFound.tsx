import { Link } from 'react-router-dom';

import { ROUTES_MAP } from 'const/routes';

const NotFound = () => (
  <div>
    Looks like you're lost. Go to{' '}
    <Link to={ROUTES_MAP.PRODUCTS}>homepage.</Link>
  </div>
);

export default NotFound;
