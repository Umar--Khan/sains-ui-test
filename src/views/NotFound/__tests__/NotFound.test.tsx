import renderer from 'react-test-renderer';

import { withRouter } from 'src/testUtils/helpers';
import NotFound from 'src/views/NotFound/NotFound';

describe('<NotFound />', () => {
  it('Should render correctly', () => {
    const component = renderer.create(withRouter(<NotFound />));
    expect(component.toJSON()).toMatchSnapshot();
  });
});
