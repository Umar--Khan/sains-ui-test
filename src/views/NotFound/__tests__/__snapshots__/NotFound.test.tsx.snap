// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> Should render correctly 1`] = `
<div>
  Looks like you're lost. Go to 
  <a
    href="/"
    onClick={[Function]}
  >
    homepage.
  </a>
</div>
`;
